packages/react-component-library/src/components/NumberInput/ClearButton.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react'
+import { IconCancel } from '@royalnavy/icon-library'
+import { selectors } from '@royalnavy/design-tokens'
+import styled, { css } from 'styled-components'
+
+const { color, spacing } = selectors
+
+interface ClearButtonProps {
+  isCondensed: boolean
+  isDisabled: boolean
+  onClick: (event: React.MouseEvent<HTMLButtonElement>) => void
+}
+
+interface StyledButtonProps {
+  isCondensed: boolean
+}
+
+const StyledButton = styled.button<StyledButtonProps>`
+  background-color: ${color('neutral', 'white')};
+  border: none;
+  color: ${color('neutral', '300')};
+  cursor: pointer;
+  margin: unset;
+  padding: unset;
+  position: relative;
+  right: ${spacing('4')};
+  width: 16px;
+  display: flex;
+  align-items: center;
```

Comment:
There were a few small styling issues to do with the focus states when I loaded up Storybook:

<img width="437" alt="Screenshot 2020-10-26 at 11 34 02" src="https://user-images.githubusercontent.com/48086589/97167706-2c82ff00-177f-11eb-993c-137fb74ad9bd.png">
<img width="443" alt="Screenshot 2020-10-26 at 11 33 55" src="https://user-images.githubusercontent.com/48086589/97167708-2d1b9580-177f-11eb-9ff6-b088290a2f5e.png">